sources/packages/web/src/router/index.ts

```diff
@@ -25,55 +34,21 @@ const router = createRouter({
   routes,
 });
 
-function forEachInstitutionRoutes(
-  to: any,
-  _from: any,
-  next: any,
-  clientType: ClientIdType,
-) {
-  // MANAGE INSTITUTION ROUTES
-  AuthService.shared
-    .initialize(clientType)
-    .then(() => {
-      if (to.meta.requiresAuth !== false) {
-        if (AuthService.shared.keycloak?.authenticated) {
-          if (to.meta?.userTypes || to.meta?.checkAllowedLocation) {
-            if (
-              UserAuthorizationService.shared.isUserTypeAllowed(
-                to.meta.userTypes,
-                to.params,
-                to.meta.checkAllowedLocation,
-              )
-            ) {
-              next();
-            } else {
-              // UNAUTHORIZED USER
-              next({
-                name: SharedRouteConst.FORBIDDEN_USER,
-              });
-            }
-          } else {
-            next();
-          }
-        } else {
-          next({
-            name: InstitutionRoutesConst.LOGIN,
-          });
-        }
-      } else {
-        next();
-      }
-    })
-    .catch((error: unknown) => {
-      console.error(error);
-      throw error;
-    });
-}
+// Define error handling on router error.
+router.onError((error: unknown) => {
+  console.error(error);
+  throw error;
+});
 
-router.beforeEach((to, from, next) => {
-  if (to.meta?.clientType === ClientIdType.Institution)
-    forEachInstitutionRoutes(to, from, next, ClientIdType.Institution);
-  else next();
+// TODO: As per vue documentation, usage of next is discouraged.
+// when vue-router version is upgraded, usage of next must be replaced
+// with returning either true or route location.
```

Comment:
As discussed it is about the usage of returning true instead of next.

![image](https://user-images.githubusercontent.com/54600590/233194727-a27886b1-8806-4a12-92fe-349059e27375.png)

Reply:
I understood after our discussion